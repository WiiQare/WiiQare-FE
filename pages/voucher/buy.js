import { createContext, useState } from 'react';
import Head from 'next/head';
import { useRouter } from 'next/router';
import DashboardLayout from '../../layouts/Dashboard';
import CardHeader from '../../components/atoms/Card/Header';
import Stepper from '../../components/atoms/Stepper';
import Step from '../../components/atoms/Stepper/step';
export const FormContext = createContext();

const Page = () => {
  const { step, redirect_status, payment_intent } = useRouter().query;
  const [activeStepIndex, setActiveStepIndex] = useState(
    step == 'end' && redirect_status == 'succeeded' ? 2 : 0,
  );
  const [formData, setFormData] = useState({});

  return (
    <>
      <Head>
        <title>Envoyer un pass santé</title>
      </Head>
      <div className="p-2 space-y-6 md:py-8 md:px-6">
        <CardHeader
          title={'Envoyer un pass santé'}
          breadcrumbs={[
            {
              item: 'Accueil',
              link: '/',
            },
            {
              item: 'Envoyer un pass santé',
              link: '/voucher/buy',
            },
          ]}
          download={false}
          print={false}
        />

        <section className="w-full pb-20 md:pb-0">
          <div className="w-full overflow-hidden md:col-span-2 rounded-lg py-8 flex flex-col gap-6 bg-white drop-shadow-sm">
            <div className="px-8 space-y-8">
              <div>
                <h2 className="font-semibold text-3xl">
                  Envoyer un pass santé
                </h2>
              </div>
              <div className="flex flex-col gap-4">
                <FormContext.Provider
                  value={{
                    activeStepIndex,
                    setActiveStepIndex,
                    formData,
                    setFormData,
                    payment_intent,
                  }}
                >
                  <div className="flex flex-col items-center justify-start">
                    <Stepper />
                    <Step />
                  </div>
                </FormContext.Provider>
              </div>
            </div>
          </div>
        </section>
      </div>
    </>
  );
};

Page.getLayout = (page) => (
  <DashboardLayout className="space-y-8">{page}</DashboardLayout>
);

export default Page;
