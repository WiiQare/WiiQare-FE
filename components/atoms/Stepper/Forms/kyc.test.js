import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import KYC from './kyc';
import { FormContext } from '../../../../pages/voucher/buy';
import { SessionProvider } from 'next-auth/react';
require('jest-fetch-mock').enableMocks();

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {
      'payment-intent': 'pi_1J4JrjGswQjYFZwX0Z1Z1Z1Z',
    },
  }),
}));

describe('KYC', () => {
  let component;
  beforeEach(() => {
    const res = render(
      <SessionProvider session={{ user: { data: { userId: 'random123' } } }}>
        <FormContext.Provider value={{ activeStepIndex: 0 }}>
          <KYC />
        </FormContext.Provider>
      </SessionProvider>
    );
    component = res.container;
  });

  it('should render the component', () => {
    expect(component).toMatchSnapshot();
  });
});