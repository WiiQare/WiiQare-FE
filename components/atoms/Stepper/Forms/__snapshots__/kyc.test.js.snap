// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KYC should render the component 1`] = `
<div>
  <div
    class="flex flex-col gap-6 justify-center items-center"
  >
    <div
      class="flex flex-col gap-2 items-center justify-center"
    >
      <div
        aria-label="loading"
        class="animate-spin inline-block w-4 h-4 border-[2px] border-current border-t-transparent text-primary rounded-full"
        role="status"
      >
        <span
          class="sr-only"
        >
          Loading...
        </span>
      </div>
      <span
        class="text-sm font-light"
      >
        Avant de continuer, nous devons vérifier votre identité...
      </span>
    </div>
  </div>
</div>
`;
